### PR TITLE
Integrations tests to check custom DAV properties persistence

### PR DESCRIPTION
**Pull request by VicDeo**
## Description
more tests

## Related Issue
https://github.com/owncloud/QA/issues/390

## Motivation and Context
Integration testing

## How Has This Been Tested?
By Behat engine

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/integration/features/dav-v2.feature [hunk 1]
```diff
@@ -80,3 +80,41 @@ Feature: dav-v2
...
 		When User "user0" uploads file "data/textfile.txt" to "/testquota/asdf.txt"
 		Then the HTTP status code should be "201"
+
+	Scenario: Setting custom DAV property and reading it
+		Given using new dav path
+		And As an "admin"
+		And user "user0" exists
+		And As an "user0"
+		And User "user0" uploads file "data/textfile.txt" to "/testcustomprop.txt"
+		And "user0" sets property of file "/testcustomprop.txt" with "{http://whatever.org/ns}very-custom-prop" "veryCustomPropValue"
```
Comment (SergioBertolinSG) [inline tests/integration/features/dav-v2.feature]: How to know what is the last string?

Maybe add "with value" before like:

And "user0" sets property of file "/testcustomprop.txt" with "{http://whatever.org/ns}very-custom-prop" with value "veryCustomPropValue"

Or if this string is too long use a tablenode
And "user0" sets property of file "/testcustomprop.txt" with
 | propertyName | propertyValue |

Optional could be lo allow to set several properties at once

And "user0" sets properties of file "/testcustomprop.txt" with
  | propertyName1 | propertyValue1 |
  | propertyName2 | propertyValue2 |

File: tests/integration/features/dav-v2.feature [hunk 2]
```diff
@@ -80,3 +80,41 @@ Feature: dav-v2
...
+		And as "user0" gets a custom property "{http://whatever.org/ns}very-custom-prop" of file "/testcustomprop.txt"
```
Comment (SergioBertolinSG) [inline tests/integration/features/dav-v2.feature]: This statement should be When, or the set property one, but there has to be a when.

File: tests/integration/features/dav-v2.feature [hunk 3]
```diff
@@ -80,3 +80,41 @@ Feature: dav-v2
...
+
+	Scenario: Setting custom DAV property and reading it after the file is renamed
+		Given using new dav path
+		And As an "admin"
+		And user "user0" exists
+		And As an "user0"
+		And User "user0" uploads file "data/textfile.txt" to "/testcustompropwithmove.txt"
+		And "user0" sets property of file "/testcustompropwithmove.txt" with "{http://whatever.org/ns}very-custom-prop" "valueForMovetest"
+		And User "user0" moved file "/testcustompropwithmove.txt" to "/catchmeifyoucan.txt"
+		And as "user0" gets a custom property "{http://whatever.org/ns}very-custom-prop" of file "/catchmeifyoucan.txt"
```
Comment (SergioBertolinSG) [inline tests/integration/features/dav-v2.feature]: Missing When.

File: tests/integration/features/dav-v2.feature [hunk 4]
```diff
@@ -80,3 +80,41 @@ Feature: dav-v2
...
+		And As an "user0"
+		And User "user0" uploads file "data/textfile.txt" to "/testcustompropshared.txt"
+		And as "user0" creating a share with
+		  | path | testcustompropshared.txt |
+		  | shareType | 0 |
+		  | permissions | 31 |
+		  | shareWith | user1 |
+		And "user0" sets property of file "/testcustompropshared.txt" with "{http://whatever.org/ns}very-custom-prop" "valueForSharetest"
+		And As an "user1"
+		And as "user1" gets a custom property "{http://whatever.org/ns}very-custom-prop" of file "/testcustompropshared.txt"
```
Comment (SergioBertolinSG) [inline tests/integration/features/dav-v2.feature]: Missing When.

**Review by SergioBertolinSG — CHANGES_REQUESTED** [review]

**Comment by VicDeo** [comment]
@SergioBertolinSG thanks for review, updated

**Review by SergioBertolinSG — APPROVED** [review]

**Comment by PVince81** [comment]
@VicDeo please rebase. JS tests env issue should be fixed.

**Comment by DeepDiver1975** [comment]
> @VicDeo please rebase. JS tests env issue should be fixed.

I already did - wanted to know if this is really fixed

**Comment by DeepDiver1975** [comment]
js tests are green!

**Pull request merged**